src/niscope/templates/session.py/fancy_fetch.py.mako

```diff
@@ -0,0 +1,43 @@
+<%page args="f, config, method_template"/>\
+<%
+    '''Forwards to _fetch_double()'''
+    import build.helper as helper
+
+    suffix = method_template['method_python_name_suffix']
+%>\
+    def ${f['python_name']}${suffix}(${helper.get_params_snippet(f, helper.ParameterUsageOptions.SESSION_METHOD_DECLARATION)}):
+        '''${f['python_name']}
+
+        ${helper.get_function_docstring(f, False, config, indent=8)}
+        '''
+        import sys
+
+        # Set the fetch attributes
+        with _NoChannel(session=self):
+            self._fetch_relative_to = relative_to
+            self._fetch_offset = offset
+            self._fetch_record_number = record_number
+            self._fetch_num_records = num_records
+            if num_samples is None:
+                num_samples = self.horz_record_length
+
+        wfm, wfm_info = self._fetch(num_samples, timeout)
+
+        if sys.version_info.major < 3:
+            # memoryview in Python 2 doesn't support numeric types, so we copy into an array.array to put in the wfm. :( You should be using Python 3!
+            # Or use the _into version. memoryview in Python 2 only supports string and bytearray, not array.array or numpy.ndarray of arbitrary types.
+            for i in range(len(wfm_info)):
+                start = i * num_samples
+                end = start + num_samples
+                wfm_info[i].wfm = array.array('d', wfm[start:end])
+        else:
+            # In Python 3 and newer we can use memoryview objects to give us pieces of the underlying array. This is much faster
+            mv = memoryview(wfm)
+
+            for i in range(len(wfm_info)):
+                start = i * num_samples
+                end = start + num_samples
+                wfm_info[i].wfm = mv[start:end]
```

Comment:
It seems most of the code in the the if / else blocks is the same except for this line and the creation of the memory view. If you think you can reduce repetition without making it dirtier, that would be preferable.